.github/workflows/formatting_python.yml

```diff
@@ -0,0 +1,44 @@
+name: Check Python formatting
+
+on:
+  push:
+    paths:
+      - 'python/**'
+      - 'cime_config/SystemTests/**'
+      - 'cime_config/buildlib/**'
+      - 'cime_config/buildnml/**'
+  pull_request:
+    paths:
+      - 'python/**'
+      - 'cime_config/SystemTests/**'
+      - 'cime_config/buildlib/**'
+      - 'cime_config/buildnml/**'
+
+jobs:
+  lint-and-format-check:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the code
+      - uses: actions/checkout@v4
+
+      # Set up the conda environment
+      - uses: conda-incubator/setup-miniconda@v3
+        with:
+          activate-environment: ctsm_pylib
+          environment-file: python/conda_env_ctsm_py.yml
+          channels: conda-forge
+          auto-activate-base: false
+
+      # Run pylint check
+      - name: Run pylint
+        run: |
+          cd python
+          conda run -n ctsm_pylib make lint
```

Comment:
Ahh, nice. This makes the workflow nicely aligned with the Makefile that we use.

I wanted to get the workflow to use the make directly, but couldn't get that to go. I think it's this technique you used here to run "make lint" and later "make black" beneath "conda run" that get's it to go here. make probably needs to be run under the conda subshell.

Reply:
Yeah, that's exactly it.